src/base/bittorrent/session.cpp

```diff
@@ -4513,6 +4514,43 @@ void Session::startUpTorrents()
         if (needStore)
              m_resumeDataStorage->store(torrentID, resumeData);
 
+        const QString category = resumeData.category;
+        bool isCategoryRecovered = recoveredCategories.contains(category);
+        if (!category.isEmpty() && (isCategoryRecovered || !m_categories.contains(category)))
+        {
+            if (!isCategoryRecovered)
+            {
+                if (addCategory(category))
+                {
+                    recoveredCategories.insert(category);
+                    isCategoryRecovered = true;
+                    LogMsg(tr("Inconsistent data is detected. Category '%2' is assigned to some torrent(s) but doesn't exist in configuration file."
+                              " Its settings will be reset to default.").arg(torrentID.toString(), category), Log::WARNING);
+                }
+                else
+                {
+                    resumeData.category.clear();
+                    LogMsg(tr("Inconsistent data is detected. Torrent '%1' has assigned to category '%2' which is invalid.")
+                           .arg(torrentID.toString(), category), Log::WARNING);
+                }
+            }
+
+            // We should check isCategoryRecovered again since the category
+            // can be just recovered by the code above
+            if (isCategoryRecovered && resumeData.useAutoTMM)
+            {
+                const Path storageLocation {resumeData.ltAddTorrentParams.save_path};
+                if ((storageLocation != categorySavePath(resumeData.category)) && (storageLocation != categoryDownloadPath(resumeData.category)))
+                {
+                    resumeData.useAutoTMM = false;
+                    resumeData.savePath = storageLocation;
+                    resumeData.downloadPath = {};
+                    LogMsg(tr("Torrent '%1' has assigned recovered category '%2' but doesn't match any category path. Torrent is switched to \"Manual\" mode.")
```

Comment:
>Torrent '%1' has assigned recovered category '%2' but doesn't match any category path.

Does this mean that the recovered category might match the category path of some other category? That's what i understood.

Reply:
> Does this mean that the recovered category might match the category path of some other category?

Sorry, my English is really bad.
It should mean that actual path of torrent doesn't match any path of category (save path and download path).

Reply:
>It should mean that actual path of torrent doesn't match any path of category (save path and download path).

Maybe this:

```suggestion
                    LogMsg(tr("Torrent '%1' has assigned recovered category '%2' whose paths don't match the torrent's path. Torrent is switched to \"Manual\" mode.")
```